### DISCUSSION TITLE
[QUESTION] How to run an entire desktop environment from distrobox

### DISCUSSION THREAD

**Discussion by LiberQuack** [89luca89/distrobox#1447] [General]
I tried to follow this post and some others to run a full desktop environment from the container
https://github.com/89luca89/distrobox/blob/main/docs/posts/run_latest_gnome_kde_on_distrobox.md#running-latest-gnome

But didn't achieve success my initial experiment was about running GNOME 45 on a Fedora 40 KDE HOST
My real goal, is to run steamOS ui from arch container

If someone (community/members) can give me some tips on how to troubleshoot/achieve it, I would appreciate it very much :)

**Comment by b3n10** [comment]
Were you able to run a full DE? Mine works okay with some caveat:

```
 distrobox create --image fedora:latest --name fullKDE --home /home/user/fedora2
 
 # login to the container and install KDE (~3GB)
 distrobox enter fullKDE
 sudo dnf environment install kde-desktop-environment
 
 # add the desktop file for the login screen and follow other steps here - https://github.com/89luca89/distrobox/blob/main/docs/posts/run_latest_gnome_kde_hyprland_on_distrobox.md
```

It logs me in and can run apps with audio and internet. The volume icon is disabled and wifi doesn't show any networks, but hardware keys can still manage the volume and put to sleep.

But the biggest flaw here is that doesn't take up the full screen. The full DE is inside a box pushed to the top left corner. So anything outside that box is just black, but my mouse can still move around the whole screen. When i check the display config, it stuck in 1024x768 even though my laptop screen can support up to 1920x1080. It cannot also the detect my external display.

<img width="4080" height="3072" alt="kde DE" src="https://github.com/user-attachments/assets/3e54092e-8e40-42b3-b2bc-880222b78b27" />

**Reply by airtonix** [reply]
This would all be to do with the `--device` flag of docker: https://stackoverflow.com/questions/34302096/sharing-devices-webcam-usb-drives-etc-with-docker

not sure how distrobox would allow you do use that...

but this is where you need to explore
